app/models/piece.rb

```diff
@@ -79,8 +80,26 @@ def invalid_input?(destination_row, destination_col)
     destination_row > 7 || destination_col > 7
   end
 
-  def invalid_destination?(destination_row, destination_col)
-    # # This has a piece in the destination, but not in between the pieces.
-    game.pieces.where(current_row_index: destination_row, current_column_index: destination_col, color: color).count > 0
+  def spot_taken?(destination_row, destination_col)
+    # This has a piece in the destination, but not in between the pieces.
+    game.pieces.where(current_row_index: destination_row, current_column_index: destination_col).count > 0
```

Comment:
Alternate way of writing this.

```
game.pieces.where(current_row_index: destination_row, current_column_index: destination_col).exists?
```

Also, you could remove the comment from this method. The method name is awesome & very descriptive. So I don't think the comment adds anything to it.